File: core/src/com/qtech/bubbleblaster/common/Progress.java
package com.qtech.bubbleblaster.common;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
public class Progress {
    private int progress;
    private final int max;

    public Progress(int max) {
        this(0, max);
    }

    public void increment() {
        if (progress + 1 <= max) {
            progress++;
        } else {
            throw new IllegalStateException("Progress increment at end: " + (progress + 1) + ", max: " + max);
        }
    }
}

File: src/main/kotlin/qtech/bubbles/common/Progress.kt
package qtech.bubbles.common

import lombok.AllArgsConstructor
import lombok.EqualsAndHashCode
import lombok.Getter
import lombok.ToString

@Getter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
open class Progress(var progress: Int, val max: Int) {

    constructor(max: Int) : this(0, max)

    fun increment() {
        if (progress + 1 <= max) {
            progress++
        } else {
            throw IllegalStateException("Progress increment at end: " + (progress + 1) + ", max: " + max)
        }
    }
}